pubpub/pubpub
# Bh activity digest emails wip

Addresses #1473 

## summary
Communities, especially their managers, would like to get email-delivered digests of developments within their communities. This PR uses the activity item system to meet that feature need. It fetches activity items for current community and renders them into email friendly HTML with inline styles, viewable at `[community base url]/email/digest`.

## target
Aiming at visual parity with [these mocks](https://www.figma.com/file/RUmAt42roXLhNobGwd9yfa/email-digest-draft?node-id=25%3A639)

## implementation
+ [styled-components](https://styled-components.com/) for inline css.
+ [juice](https://github.com/Automattic/juice#readme) to to inline css properties to the style attribute for maximum email friendliness
+ a lot of activityItem sorting/butchery

## still to complete
+ send these puppies out with mailgun
+ set up chron job to run weekly
+ subscription management toggle in dash
+ digest settings to manage type/volume of activity items in digest
+ email header/metadata
+ any kind of tests
+ currently, operates for current logged in user/community only -> wire up for arbitrary community

## attention reviewers!
+ the topmost component background/color/text is unreliable from community to community. I'm taking all suggestions for how to manage the different results visible between, for example, the email for the Contours community vs Fermentology vs Qualitative Criminology
+ how to capitalize the 'y' in `you created a Release of this pub` when you are forbidden from using css classes?
+ more reusable component breakdown for future email templating
+ CSS best practices, as I was merely shooting for visual match

## testing
Choose any three communities that have regular activity for the last week. At a minimum, check out the [demo community](demo.duqduq.org). For each of them, walk through the following:
+ log in to a community admin account
+ access each of their community activity dashboards
+ in parallel window, pull up the digest email preview by navigating to `[community base url]/email/digest`. If in dev. environment, modify the `PUBPUB_LOCAL_COMMUNITY` variable to, for example, `fermentology` then access `localhost:9876/email/digest` to see the rendered fermentology digest email. If on duqduq, you would go to fermentology.duqduq.org/email/digest
+ confirm that the all activity items from within the previous week are present in the digest
+ confirm that the activity items are grouped  by the affected object (pub/community/collection/page)
+ proof rendering of copy and design parity with the figma link above.

## questions
+ should we limit the number of ActivityItems returned? It's currently (arbitrarily) set to 150 items
